ui/src/views/plugins/ApiDocsPlugin.vue

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+<template>
+  <div>
+    <resource-layout>
+      <template #left>
+        <a-card :bordered="false">
+          <a-auto-complete
+            v-model:value="query"
+            :options="options"
+            style="width: 100%"
+            >
+            <a-input-search
+                size="default"
+                :placeholder="$t('label.search')"
+                v-model:value="query"
+                allow-clear
+                enter-button
+                >
+                <template #prefix><search-outlined /></template>
+            </a-input-search>
+          </a-auto-complete>
+          <a-list style="margin-top: 12px; height:580px; overflow-y: scroll;" size="small" :data-source="Object.keys($store.getters.apis).sort()">
+            <template #renderItem="{ item }">
+              <a>
+              <a-list-item v-if="item.toLowerCase().includes(query.toLowerCase())" @click="showApi(item)">
+                <span v-if="selectedApi === item">
+                  <strong>{{ item }}</strong> <a-tag v-if="$store.getters.apis[item].isasync" color="blue">async</a-tag>
+                </span>
+                <span v-else>
+                  {{ item }} <a-tag v-if="$store.getters.apis[item].isasync" color="blue">async</a-tag>
+                </span>
+              </a-list-item>
+              </a>
+            </template>
+          </a-list>
+          <a-divider style="margin-bottom: 12px" />
+          <span>{{ Object.keys($store.getters.apis).length }} {{ $t('label.api.docs.count') }}</span>
```

Comment:
there are 700+ APIs, pagination possible in the API listing?

Reply:
Pagination would make it weird, there’s a search bar which maybe more useful